src/Init/Data/Option/Lemmas.lean

```diff
@@ -374,9 +374,15 @@ end choice
 
 -- See `Init.Data.Option.List` for lemmas about `toList`.
 
-@[simp] theorem or_some : (some a).or o = some a := rfl
+@[simp] theorem some_or : (some a).or o = some a := rfl
 @[simp] theorem none_or : none.or o = o := rfl
 
+@[deprecated some_or] theorem or_some : (some a).or o = some a := rfl
```

Comment:
We usually add a deprecation date here. For example like here https://github.com/leanprover/lean4/blob/cd24e9dad4987ca127f58b8cd2d041fc625173f2/src/Lean/Data/HashSet.lean#L222-L226